os/pm/pm_activity.c

```diff
@@ -202,6 +208,36 @@ void pm_stay(int domain, enum pm_state_e state)
 	leave_critical_section(flags);
 }
 
+/****************************************************************************
+ * Name: pm_set_timer
+ *
+ * Description:
+ *   This function is called to set a timed callback to an intended function.
+ *   It may be used to invoke pm_relax() after a fixed time duration of
+ *   locking using pm_stay().
+ *
+ * Input Parameters:
+ *   timer_type - the type of the timer which determines the callback
+ *   timer_interval - duration of the timer
+ *
+ * Returned Value:
+ *   None.
+ *
+ ****************************************************************************/
+
+void pm_set_timer(int pm_timer_type, size_t timer_interval)
+{
+	/* Set the global variables to trigger the appropriate timer */
+	g_pm_timer.timer_type = pm_timer_type;
+	g_pm_timer.timer_interval = timer_interval;
```

Comment:
Is this required to do only one time set? If yes then how are we restricting? If no then what will be the side effects of updating in multiple times?

Reply:
Multiple synchronous timers are not supported right now.